src/golang/lib/lib_utils/utils.go

```diff
@@ -113,3 +114,35 @@ func ParseDatabricksConfig(conf auth.Config) (*integration.DatabricksIntegration
 
 	return &c, nil
 }
+
+func ParseEmailConfig(conf auth.Config) (*integration.EmailConfig, error) {
+	data, err := conf.Marshal()
+	if err != nil {
+		return nil, err
+	}
+
+	var c struct {
+		User              string                   `json:"user" yaml:"user"`
+		Password          string                   `json:"password" yaml:"password"`
+		Host              string                   `json:"host" yaml:"host"`
+		Port              string                   `json:"port" yaml:"port"`
+		TargetsSerialized string                   `json:"targets_serialized" yaml:"targets_serialized"`
+		Level             shared.NotificationLevel `json:"level"`
+	}
+	if err := json.Unmarshal(data, &c); err != nil {
+		return nil, err
+	}
+
+	var targets []string
+	if err := json.Unmarshal([]byte(c.TargetsSerialized), &targets); err != nil {
```

Comment:
Is it possible to do all the unmarshalling in one step if `TargetSerialized` in the struct above is of type `[]string`?

Reply:
No, since the config stores a `map[string]string`, the raw json value will be `{..., "target_serialized": "[\"addr\"]"}` which requires another layer of deserialization. The issue comes from the fact that a config value must be a string, and we need the conversion if we want to use it as something else

Reply:
makes sense!